datanode/api/trading_data_v2.go

```diff
@@ -5559,3 +5559,89 @@ func (t *TradingDataServiceV2) ListAMMs(ctx context.Context, req *v2.ListAMMsReq
 		},
 	}, nil
 }
+
+func (t *TradingDataServiceV2) EstimateAMMBounds(ctx context.Context, req *v2.EstimateAMMBoundsRequest) (*v2.EstimateAMMBoundsResponse, error) {
+	defer metrics.StartAPIRequestAndTimeGRPC("EstimateAMMBounds")()
+
+	if req.MarketId == "" {
+		return nil, formatE(ErrInvalidMarketID)
+	}
+
+	basePrice, overflow := num.UintFromString(req.BasePrice, 10)
+	if overflow || basePrice.IsNegative() {
+		return nil, formatE(ErrInvalidBasePrice)
+	}
+
+	upperPrice, overflow := num.UintFromString(req.UpperPrice, 10)
+	if overflow || upperPrice.IsNegative() {
+		return nil, formatE(ErrInvalidUpperPrice)
+	}
+
+	lowerPrice, overflow := num.UintFromString(req.LowerPrice, 10)
```

Comment:
Don't know whose taking over this PR but we should validate that `lower price < base price < upper price` otherwise the maths is going to go wild.

Reply:
checks added